ibacm/src/acm.c

```diff
@@ -2620,6 +2620,27 @@ static void acm_open_dev(struct ibv_device *ibdev)
 		goto err1;
 	}
 
+	for (i = 0; i < attr.phys_port_cnt; i++) {
+		ret = ibv_query_port(verbs, i + 1, &port_attr);
+		if (ret) {
+			acm_log(0, "ERROR - ibv_query_port (%s, %d) return (%d)\n",
+				ibdev->name, i + 1, ret);
+			continue;
+		}
+
+		if (port_attr.link_layer == IBV_LINK_LAYER_INFINIBAND) {
+			acm_log(1, "%s port %d is an InfiniBand port\n", ibdev->name, i + 1);
+			has_ib_port = true;
+		} else {
+			acm_log(1, "%s port %d is not an InfiniBand port\n", ibdev->name, i + 1);
+		}
```

Comment:
These six lines can of course be written as:

has_ib_port = port_attr.link_layer == IBV_LINK_LAYER_INFINIBAND;
acm_log(1, "%s port %d is %san InfiniBand port\n", ibdev->name, i + 1, has_ib_port ? "" : "not ");

but I am OK with this either way.